src/test/common_url_tests.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2024 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <test/util/setup_common.h>
+#include <common/url.h>
+
+#include <string>
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_AUTO_TEST_SUITE(common_url_tests)
+
+// These test vectors were ported from test/regress.c in the libevent library
+// which used to be a dependency of the UrlDecode function.
```

Comment:
nit: it used to be a dependency of `urlDecode`, not `UrlDecode`, but I guess it's fine

Reply:
Either way, someone might get confused, but I think it's better to keep it as is.